src/coreclr/jit/target.h

```diff
@@ -208,12 +208,164 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+typedef _regNumber_enum regNumber;
+typedef unsigned char   regNumberSmall;
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct regMaskTP
+{
+private:
+    uint64_t low;
+
+public:
+    constexpr regMaskTP(uint64_t regMask)
+        : low(regMask)
+    {
+    }
+
+    regMaskTP()
+    {
+    }
+
+    FORCEINLINE explicit operator bool() const
+    {
+        return low != RBM_NONE;
+    }
+
+    FORCEINLINE explicit operator regMaskSmall() const
+    {
+        return (regMaskSmall)low;
+    }
+
+    FORCEINLINE explicit operator unsigned int() const
+    {
+        return (unsigned int)low;
+    }
+
+    FORCEINLINE uint64_t getLow() const
+    {
+        return low;
+    }
+
+    FORCEINLINE void setLow(uint64_t _low)
+    {
+        low = _low;
+    }
```

Comment:
I don't think we should have this one (can just assign the full `regMaskTP` instead)

Reply:
(FWIW, it seems like bad encapsulation to have `low` accessible outside this class at all)

Reply:
I totally agree with you and was not a fan of having this in the last commit. May be I should just try assigning the whole `regMaskTP`.